projects/table-builder/src/lib/classes/data-filter.ts

```diff
@@ -1,12 +1,13 @@
 import { Observable, combineLatest } from 'rxjs';
-import { map, shareReplay } from 'rxjs/operators';
+import { map, refCount, publishReplay } from 'rxjs/operators';
 
 export class DataFilter {
   filteredData$: Observable<any[]>;
   constructor(filters$: Observable< Array< (val: any) => boolean>>, data$: Observable<any[]>) {
     this.filteredData$ = combineLatest([data$, filters$]).pipe(
       map(([data, filters]) => this.filter(data, filters)),
-      shareReplay()
+      publishReplay(1),
```

Comment:
So what is the difference between `shareReplay` and publishReplay(1)` with `refCount`

Reply:
shareReplay({refCount: true}) can sometimes resubscribe which can be a problem with http requests.